pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/ClustersBase.java

```diff
@@ -771,8 +781,13 @@ private CompletableFuture<Void> filterAndUnloadMatchedNamespaceAsync(NamespaceIs
                             .map(tenant -> adminClient.namespaces().getNamespacesAsync(tenant));
                     return FutureUtil.waitForAll(completableFutureStream)
                             .thenApply(namespaces -> {
-                                // if namespace match any policy regex, add it to ns list to be unload.
+                                // Filter namespaces that have current cluster in their replication_clusters
+                                // if namespace match any policy regex, add it to ns list to be unloaded.
                                 return namespaces.stream()
+                                        .filter(namespaceName -> adminClient.namespaces()
+                                                .getPoliciesAsync(namespaceName)
+                                                .thenApply(policies -> policies.replication_clusters.contains(cluster))
+                                                .join())
                                         .filter(namespaceName ->
                                                 policyData.getNamespaces().stream().anyMatch(namespaceName::matches))
```

Comment:
this should be the first filter so that policies aren't retrieved unnecessarily

Reply:
Addressed in https://github.com/apache/pulsar/pull/23100